cfme/fixtures/base.py

```diff
@@ -58,3 +58,19 @@ def fix_missing_hostname(appliance):
 
             if ssh_client.run_command('grep $(hostname) /etc/hosts').failed:
                 logger.error('Failed to mangle /etc/hosts')
+
+
+@pytest.fixture(scope="session", autouse=True)
+def fix_missing_appliance_name(appliance):
+    """Fix for an empty appliance server name"""
+    if isinstance(appliance, DummyAppliance) or appliance.is_dev:
+        return
+    logger.info('Checking appliance server name is set')
+    if appliance.server.name == '':
```

Comment:
Q: Why can't you just do?

```python,
if not appliance.server.name:
```

Reply:
can, just being as explicit as possible.